server/auth.go

```diff
@@ -22,24 +22,31 @@ var AuthorizedUsers gin.Accounts = gin.Accounts{
 	"john@guest.com": "guest",
 }
 
+// SimpleUserDB models a database. Users will be stored in a DB in the future
 var SimpleUserDB map[string]User = map[string]User{
 	"dan@admin.com":  {ID: 1, Email: "dan@admin.com", AccountType: "admin"},
 	"john@guest.com": {ID: 1, Email: "john@guest.com", AccountType: "guest"},
 }
 
+// LoginRequest requires login requests to have an email and password.
 type LoginRequest struct {
 	Email    string `json:"email"`
 	Password string `json:"password"`
 }
 
-func (app *App) Login(c *gin.Context) {
-	byteData, err := ioutil.ReadAll(c.Request.Body)
+/* Login authenticates the user and returns a user ID and a token
+*  Login is a function of app */
+func (app *app) Login(c *gin.context) {
+	// Read the request from the gin context
+	bytedata, err := ioutil.readall(c.request.body)
```

Comment:
this `readall` isn't called correctly. Should be `ReadAll`

Reply:
and I think it should be `c.Request.Body` as well

Reply:
fixed mistakes @mikelu92 login still does not work. Can you take a look?